extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -467,7 +465,30 @@ public function getCuratedContentSection() {
 			throw new NotFoundApiException( 'No members' );
 		}
 
-		$this->response->setVal( 'items', $data[ 'items' ] );
+		$this->response->setVal( 'items', $data['items'] );
+	}
+
+	public function getMainPageDetailsAndAdsContext() {
+		$mainPageTitle = Title::newMainPage();
+		$mainPageArticleID = $mainPageTitle->getArticleID();
+		$article = Article::newFromID( $mainPageArticleID );
+
+		$articleDetails = $this->getArticleDetails( $article );
+
+		if ( empty( $articleDetails ) ) {
+			throw new WikiaException( 'Article details empty', 500 );
+		}
+
+		$adsContext = $this->mercuryApi->getAdsContext( $mainPageTitle );
+
+		if ( empty( $adsContext ) ) {
```

Comment:
We already handle this scenario in articles. If ad engine is disabled and `adsContext` is empty we just don't display ads. This isn't an error that should prevent us from providing main page details to the SPA, it's just a flag controlling ads that's turned off on some wikis.